django/utils/tree.py

```diff
@@ -70,6 +70,12 @@ def __eq__(self, other):
             return self.children == other.children
         return False
 
+    def __hash__(self):
+        node_hash = hash(self.__class__) ^ hash(self.connector) ^ self.negated
+        for child in self.children:
+            node_hash ^= hash(child)
+        return node_hash
```

Comment:
I didn't check this carefully, but I don't think this can be an issue because `__hash__` is not called very often. As far as I'm concerned `__hash__` should be consistent with `__eq__` which compares `children`.